diff -Nuar --exclude '*.orig' awesome-3.5-rc1.orig/awesomeConfig.cmake awesome-3.5-rc1/awesomeConfig.cmake
--- awesomeConfig.cmake	2012-11-24 13:51:33.000000000 +0000
+++ awesomeConfig.cmake	2012-11-24 20:42:23.743412441 +0000
@@ -55,7 +55,7 @@
     a_find_program(LDOC_EXECUTABLE ldoc.lua FALSE)
 endif()
 # theme graphics
-a_find_program(CONVERT_EXECUTABLE convert TRUE)
+a_find_program(CONVERT_EXECUTABLE bin/convert TRUE)
 # pkg-config
 include(FindPkgConfig)
 # lua
