--- razorstatus.c	2014-11-01 23:37:55.726213952 +0800
+++ razorstatus.c	2014-11-01 23:35:53.498211163 +0800
@@ -1,4 +1,8 @@
 #define _BSD_SOURCE
+//BAT0 begin
+#define STATUS_FILE     "/sys/class/power_supply/BAT0/status"
+#define CAPACITY_FILE   "/sys/class/power_supply/BAT0/capacity"
+//BAT0 end
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -91,6 +95,22 @@ loadavg(void)
 	return smprintf("%.2f %.2f %.2f", avgs[0], avgs[1], avgs[2]);
 }
 
+
+int
+open_capacity(void) {
+        FILE *capacity_file;
+	int capacity;
+        capacity_file = fopen(CAPACITY_FILE, "r");
+        if ( capacity_file == NULL ) {
+                fprintf(stderr, "Failed to open: %s\n", CAPACITY_FILE);
+                exit(EXIT_FAILURE);
+        }
+        fscanf(capacity_file, "%d", &capacity);
+        fclose(capacity_file);
+        return capacity;
+}
+
+
 int
 main(void)
 {
@@ -98,6 +118,7 @@ main(void)
 	char *avgs;
 	char *tmutc;
 	char *tmsh;
+	int batlevel;
 
 	if (!(dpy = XOpenDisplay(NULL))) {
 		fprintf(stderr, "razorstatus: cannot open display.\n");
@@ -108,13 +129,15 @@ main(void)
 		avgs = loadavg();
 		tmutc = mktimes("%H:%M", tzutc);
 		tmsh = mktimes("KW %W %c", tzshanghai);
+		batlevel = open_capacity();
 
-		status = smprintf("[LA](%s) [UTC](%s) [CST](%s)",
-				avgs, tmutc, tmsh);
+		status = smprintf("[LA](%s) [UTC](%s) [CST](%s) [BAT](%d)",
+				avgs, tmutc, tmsh, batlevel);
 		setstatus(status);
 		free(avgs);
 		free(tmutc);
 		free(tmsh);
+//		free(batlevel);
 		free(status);
 	}
 
